### PR TITLE
Fix py37 support

### PR DESCRIPTION
**Pull request by bennettgoble**
Remove a few 3.8+ changes so that autobuild continues to function on 3.7. This is needed as long as LL is using Debian Buster.

### CODE REVIEW

File: autobuild/common.py [hunk 1]
```diff
@@ -326,12 +326,12 @@ def compute_hash(path: str, hash: Callable[[], hashlib._Hash]):
...
-    b = bytearray(128*1024)
-    mv = memoryview(b)
     try:
         with open(path, 'rb') as f:
-            while n := f.readinto(mv):
-                h.update(mv[:n])
+            chunk = f.read(8192)
+            while chunk:
+                h.update(chunk)
+                chunk = f.read(8192)
```
Comment (nat-goodspeed) [inline autobuild/common.py:334]: I'm good with this change, but I do notice the chunk size changed from 128K to 8K -- intentional?

File: tests/test_install.py [hunk 2]
```diff
@@ -751,7 +751,8 @@ def test_download(self, mock_urlopen: MagicMock):
         with envvar("AUTOBUILD_GITHUB_TOKEN", None):
             autobuild_tool_install.download_package("https://example.org/foo.tar.bz2")
             mock_urlopen.assert_called()
-            got_req = mock_urlopen.mock_calls[0].args[0]
+            _, args, _ = mock_urlopen.mock_calls[0]
+            got_req = args[0]
```
Comment (nat-goodspeed) [inline tests/test_install.py:755]: Is this a Python-version-dependent change to the mock API? I don't understand why these changes are needed for backing off to support Python 3.7.

Also, couldn't this be more succinctly expressed as:

    got_req = mock_urlopen.mock_calls[0][1][0]

I guess that's less clear though.

Comment (bennettgoble) [reply tests/test_install.py:755]: These tests were failing on 3.7 due to differences in the mocking API between 37 and later version. The destructuring was done for clarity and to resolve the incompatibility.

File: autobuild/common.py (hunk 1 above)
Comment (bennettgoble) [reply autobuild/common.py:334]: Haven't bothered to benchmark, but went with smaller/common since it's no longer using a memoryview. I wasn't interested in playing golf and wanted to KISS.

File: tests/test_install.py (hunk 2 above)
Comment (nat-goodspeed) [reply tests/test_install.py:755]: > differences in the mocking API between 37 and later version

Thanks, that's what I wanted to know. Presumably later versions are also compatible with the 3.7 API.

**Review by nat-goodspeed — APPROVED** [review]
